### ISSUE TITLE
DataSearch: Change query

### ISSUE THREAD

**Issue by siddharthlatest** [appbaseio/reactivebase#92]
Needs to be a combination of phrase_prefix + boolean query.

Sample:

```
{
  "query": {
    "bool": {
      "should": [
        {
          "multi_match": {
            "query": "ycombinat",
            "fields": [
              "text",
              "title",
              "by"
            ],
            "type": "phrase_prefix"
          }
        },
        {
          "multi_match": {
            "query": "ycombinat",
            "fields": [
              "title",
              "text",
              "by"
            ],
            "type": "boolean",
            "minimum_should_match": "50%"
          }
        }
      ],
      "minimum_should_match": "1"
    }
  },
  "highlight": {
    "fields": {
      "text": {},
			"title": {}
    }
  }
}
```

**Comment by farhan687** [comment]
Fixed here: a48b1b95b31b0d6bfe35dbb62b663f0f3c8896ba

**Comment by siddharthlatest** [comment]
Fixed here: ea3b56def02515d8fee1e6f19e59f1605f2b4cb8.
